app/oauth.go

```diff
@@ -574,19 +574,14 @@ func (a *App) LoginByOAuth(service string, userData io.Reader, teamId string) (*
 		return nil, model.NewAppError("LoginByOAuth", "api.user.login_by_oauth.parse.app_error",
 			map[string]interface{}{"Service": service}, "", http.StatusBadRequest)
 	}
-	authUser := provider.GetUserFromJson(bytes.NewReader(buf.Bytes()))
 
-	authData := ""
-	if authUser.AuthData != nil {
-		authData = *authUser.AuthData
-	}
-
-	if len(authData) == 0 {
+	authUser, err1 := provider.GetUserFromJson(bytes.NewReader(buf.Bytes()))
+	if err1 != nil || *authUser.AuthData == "" {
 		return nil, model.NewAppError("LoginByOAuth", "api.user.login_by_oauth.parse.app_error",
 			map[string]interface{}{"Service": service}, "", http.StatusBadRequest)
```

Comment:
We're losing the error details.  Not sure how useful but easy enough to include them but only if err1 != nil